src/client/snd_dma.c

```diff
@@ -773,6 +773,55 @@ void S_Base_StartSoundEx(vec3_t origin, int entnum, int entchannel, sfxHandle_t
 	ch->doppler     = qfalse;
 }
 
+/**
+ * @brief Stops an already playing sound.
+ *
+ * @param[in] origin
+ * @param[in] entnum
+ * @param[in] entchannel
+ * @param[in] sfxHandle
+ * @param[in] flags
+ * @param[in] volume
+ */
+void S_Base_StopSounds(vec3_t origin, int entnum, int entchannel, sfxHandle_t sfxHandle, int flags)
+{
+	int   i;
+	sfx_t *sfx;
+
+	sfx = &knownSfx[sfxHandle];
+
+	if (sfx->inMemory == qfalse)
+	{
+		S_memoryLoad(sfx);
+	}
+
+	if (s_show->integer == 1)
+	{
+		Com_Printf("S_Base_StopSounds: %i : %s\n", s_paintedtime, sfx->soundName);
+	}
+
+	// shut off other sounds on this channel if necessary
+	for (i = 0; i < MAX_CHANNELS ; i++)
+	{
+		if (entchannel == CHAN_AUTO)
+		{
+			if (s_channels[i].entnum == entnum)
+			{
+				s_channels[i].thesfx = NULL;
+				// S_ChannelFree(&s_channels[i]);
```

Comment:
I also seem to have misunderstood what `S_ChannelFree` does, because using it made it crash when I: `reload mp40 - selfkill - spawn with mg42 - shoot with mg42`.

Setting `.thesfx = NULL` seems to work however.

```
Thread 1 "etl.x86_64" hit Catchpoint 2 (signal SIGSEGV), S_ChannelMalloc (time=1252352) at /home/vor/workspace/etlegacy/src/client/snd_dma.c:245
245		freelist     = *(channel_t **)freelist;
[GDB] Caught signal SIGSEGV, Segmentation fault!
#0  S_ChannelMalloc (time=1252352) at /home/vor/workspace/etlegacy/src/client/snd_dma.c:245
#1  0x00005555555e7acd in S_Base_StartSoundEx (origin=0x7fffffffd99c, entnum=-1, entchannel=0, sfxHandle=687, flags=0, volume=64)
    at /home/vor/workspace/etlegacy/src/client/snd_dma.c:701
#2  0x00005555555e7ec2 in S_Base_StartSound (origin=0x7fffffffd99c, entnum=-1, entchannel=0, sfxHandle=687, volume=64)
    at /home/vor/workspace/etlegacy/src/client/snd_dma.c:831
...
```